ladybug/datacollection.py

```diff
@@ -550,6 +445,53 @@ def _check_analysis_period(self, analysis_period):
             'Collection header. {} != {}'.format(
                 analysis_period.is_leap_year, self.header.analysis_period.is_leap_year)
 
+    def _time_interval_operation(self, interval, operation, percentile=0):
+        """Get a collection of a certain time interval with a given math operation."""
+        # retrive the function that correctly describes the operation
+        if operation == 'average':
+            funct = self._average
+        elif operation == 'total':
+            funct = self._total
+        else:
+            assert 0 <= percentile <= 100, \
+                'percentile must be between 0 and 100. Got {}'.format(percentile)
+            funct = self._get_percentile_function(percentile)
+
+        # retrive the data that correctly describes the time interval
+        if interval == 'monthly':
+            data_dict = self.group_by_month()
+            dates = self.header.analysis_period.months_int
+        elif interval == 'daily':
+            data_dict = self.group_by_day()
+            dates = self.header.analysis_period.doys_int
+        elif interval == 'monthlyperhour':
+            data_dict = self.group_by_month_per_hour()
+            dates = self.header.analysis_period.months_per_hour
+
```

Comment:
I know this method is intended to be only used inside the class but it might be still useful to add an else and throw an error if interval is invalid.

```suggestion
            else:
                raise ValueError('Invalid input value for interval: {}'.format(interval))
```